### DISCUSSION TITLE
How to preview

### DISCUSSION THREAD

**Discussion by drbobbyduke** [thegetty/quire#887] [Get Help]
When I try to preview, I get the following--something seems amiss:

rrduke@rrduke170193 washington-pentateuch % quire preview
[CLI] running eleventy using lib/11ty cli
[CLI:11ty] running eleventy serve
dyld: lazy symbol binding failed: Symbol not found: _pthread_jit_write_protect_supported_np
  Referenced from: /Users/rrduke/washington-pentateuch/node_modules/sharp/build/Release/../.././vendor/8.13.3/darwin-x64/lib/libvips-cpp.42.dylib
  Expected in: /usr/lib/libSystem.B.dylib

**Comment by Erin-Cecele** [comment]
Hi @drbobbyduke, thank you for posting this question. I have a few clarifying questions: do you get this error message after you've made any changes or just when you try to preview your project for the first time? If you ignore that error message, does the preview keep building? Or do you eventually see something that says ERR or ERROR? Also, can you confirm you've installed the LTS version of Node.js?

**Reply by drbobbyduke** [reply]
I tried again with a new project by attempting to remove all the data from the my-project template (removing all the info the New Deal Photography template).  This time, I received the following when I tried to preview some content and photos I added.
------
rrduke@Roberts-MacBook-Air Megiddo % quire preview
[CLI] running eleventy using lib/11ty cli
[CLI:11ty] running eleventy serve
[quire] DEBUG	 Figures:ImageProcessor         
      inputRoot: /Users/rrduke/Megiddo/content/_assets/images
      outputRoot: /Users/rrduke/Megiddo/_site
    
[11ty] Problem writing Eleventy templates: (more in DEBUG output)
[11ty] Cannot read properties of undefined (reading 'output') (via TypeError)
[11ty] 
[11ty] Original error stack trace: TypeError: Cannot read properties of undefined (reading 'output')
[11ty]     at get staticInlineFigureImage [as staticInlineFigureImage] (/Users/rrduke/Megiddo/_plugins/figures/figure/index.js:218:87)
[11ty]     at Figure.adapter (/Users/rrduke/Megiddo/_plugins/figures/figure/index.js:251:37)
[11ty]     at /Users/rrduke/Megiddo/_plugins/figures/index.js:46:42
[11ty]     at Array.map (<anonymous>)
[11ty]     at AsyncEventEmitter.<anonymous> (/Users/rrduke/Megiddo/_plugins/figures/index.js:46:15)
[11ty]     at async Promise.all (index 1)
[11ty]     at async Eleventy.executeBuild (/Users/rrduke/Megiddo/node_modules/@11ty/eleventy/src/Eleventy.js:1176:7)
[11ty]     at async Eleventy.watch (/Users/rrduke/Megiddo/node_modules/@11ty/eleventy/src/Eleventy.js:1014:18)
[11ty] Wrote 0 files in 0.01 seconds (v2.0.1)
rrduke@Roberts-MacBook-Air Megiddo % 
-------
Any thoughts would help.  Is there a blank "New Project" way to start so I don't have to erase the New Deal material with the chance of leaving something that causes an error.

**Reply by drbobbyduke** [reply]
Sorry for the font being so big...don't know how that happen.

**Reply by Erin-Cecele** [reply]
Can you either [upload the project to GitHub](https://quire.getty.edu/resources/github/) and share the repo with me or can you send the project to me as a zip file via WeTranser so I can find what is causing the break?

**Comment by drbobbyduke** [comment]
Erin,

I realized that I was using capital letters in ID and other places that was causing issues.  I was able to get a test project output to preview and PDF.  I would like to know from users how people start their projects.  Do they start with a completely empty folder that they populate or take them template and then just exchange out info.  Are there templates for each of the various .md files and a blank publication.yaml file.  I took the one provided and erased all the info but kept finding items I missed.

Thanks,
Bobby

ROBERT (BOBBY) DUKE
Director, Scholars Initiative | Museum of the Bible
Mobile: 909-214-6847
Email: ***@***.******@***.***>
***@***.***


From: Erin Cecele Dunigan ***@***.***>
Date: Tuesday, February 13, 2024 at 12:11 PM
To: thegetty/quire ***@***.***>
Cc: Bobby Duke ***@***.***>, Mention ***@***.***>
Subject: Re: [thegetty/quire] How to preview (Discussion #887)

Can you either upload the project to GitHub<https://urldefense.com/v3/__https:/quire.getty.edu/resources/github/__;!!IuDt8Cjgfceu!MukPrsKTsYzY1P2P6NbvNtYdkDsMr1X-UqF6H3d62LlMVOCmbjztitZpnPlNUq8oO2cquQo6POhpewrRIrerp4zpsDk$> and share the repo with me or can you send the project to me as a zip file via WeTranser so I can find what is causing the break?

**Comment by Erin-Cecele** [comment]
Hi @drbobbyduke, I'm glad you could figure out what was breaking the preview. We have a [ticket](https://github.com/thegetty/quire/issues/869) open to address the error messaging around this issue so there isn't so much guesswork in the future. 

I would say that most people (including us at Getty) begin with the default starter content and delete and replace it with the new content. But you are correct in that that approach can sometimes cause a chain reaction of errors. Oftentimes, in Quire, things need to be done in a certain order; otherwise, the preview will break. It's on my list to find a way to communicate the ideal order of operations. We have also been considering creating a new, completely blank starter template.  I would be curious to get other community feedback on this, and I will start a separate thread!
